api/envoy/config/bootstrap/v2/bootstrap.proto

```diff
@@ -228,3 +234,55 @@ message Runtime {
   // encoding <config_runtime_proto_json>`.
   google.protobuf.Struct base = 4;
 }
+
+message RuntimeLayer {
+  // :ref:`Static runtime <config_runtime_bootstrap>` layer.
+  message StaticLayer {
+    // This follows the :ref:`runtime protobuf JSON representation encoding
+    // <config_runtime_proto_json>`.
+    google.protobuf.Struct value = 1;
+  }
+
+  // :ref:`Disk runtime <config_runtime_local_disk>` layer.
+  message DiskLayer {
+    // Specifies the subdirectory to load within the :ref:`root directory
+    // <envoy_api_field_config.bootstrap.v2.LayeredRuntime.symlink_root>`. This
+    // is useful if multiple systems share the same delivery mechanism. Envoy
+    // configuration elements can be contained in a dedicated subdirectory.
+    string subdirectory = 1;
+
+    // :ref:`Append <config_runtime_local_disk_overrides>` the service cluster
+    // to the path.
+    bool append_service_cluster = 2;
+  }
+
+  // :ref:`Admin console runtime <config_runtime_admin>` layer.
+  message AdminLayer {
+  }
+
+  // Descriptive name for the runtime layer. This is only used for the runtime
+  // :http:get:`/runtime` output.
+  string name = 1;
+
+  oneof layer_specifier {
+    google.protobuf.Struct static_layer = 2;
+    DiskLayer disk_layer = 3;
+    AdminLayer admin_layer = 4;
```

Comment:
What does it mean to have multiple admin layers? Should we block that in code? If it's already blocked can we say that in the docs above?

Reply:
It means that earlier admin layers are effectively ignored, since later layers will overlay them idempotently. I don't think it's too confusing, most users will only configure a single layer.

The one quirky thing I'd be interested on hearing about here is how we handle the situation where the user decides to have no admin layer whatsoever. They will silently have no effect from the admin console.

Maybe it would make sense to bake in the admin layer as highest priority and always provide it? It seems essentially an operator break glass, so I wouldn't be too concerned not to provide this as an option.

Reply:
IMO it makes sense for the operator to be able to configure where it lands, even if it is the highest priority, but I could go either way on it. If we keep it, my thinking is:
1) Error if it's specified more than once.
2) If not specified, maybe somehow have the admin endpoint return an error?

WDYT?

Reply:
Sure, I can do something like this.